### PR TITLE
feat(transport): add support for setting quota project with envvar

### PR DESCRIPTION
**Pull request by codyoss**
Fixes: https://github.com/googleapis/google-api-go-client/issues/1891

### CODE REVIEW

File: internal/creds_test.go [hunk 1]
```diff
@@ -212,8 +212,8 @@ func TestQuotaProjectFromCreds(t *testing.T) {
 	if err != nil {
 		t.Fatalf("got %v, wanted no error", err)
 	}
-	if want, got := "", QuotaProjectFromCreds(cred); want != got {
-		t.Errorf("QuotaProjectFromCreds(validServiceAccountJSON): want %q, got %q", want, got)
+	if want, got := "", GetQuotaProject(cred, ""); want != got {
+		t.Errorf("QuotaProjectFromCreds(validServiceAccountJSON, \"\"): want %q, got %q", want, got)
```
Comment (quartzmo) [inline internal/creds_test.go:216]: Update test function name and error messages from `QuotaProjectFromCreds` to `GetQuotaProject`.

File: internal/creds.go [hunk 2]
```diff
@@ -131,14 +134,22 @@ func selfSignedJWTTokenSource(data []byte, ds *DialSettings) (oauth2.TokenSource
...
+func GetQuotaProject(creds *google.Credentials, clientOpt string) string {
+	if clientOpt != "" {
+		return clientOpt
+	}
+	if env := os.Getenv(quotaProjectEnvVar); env != "" {
+		return env
+	}
+	if creds == nil {
+		return ""
+	}
```
Comment (quartzmo) [inline internal/creds.go:148]: Do the 3 conditionals above have unit test coverage?

Comment (codyoss) [reply internal/creds.go:148]: they do now :)

Comment (codyoss) [reply internal/creds.go:148]: wait need to add one more!

Comment (codyoss) [reply internal/creds.go:148]: done

**Review by quartzmo — APPROVED** [review]

**Pull request merged**